### ISSUE TITLE
Is there an immutables Meta-Annotation

### ISSUE THREAD

**Issue by msmerc** [immutables/immutables#1301]
I'm trying to consolidate some my annotations like so:

```java
@Retention(RetentionPolicy.RUNTIME)
@JacksonAnnotationsInside
@JsonProperty("_id")
@JsonInclude(JsonInclude.Include.NON_NULL)
//Unfortunately immutables has no equivalent of @JacksonAnnotationsInside.
@Nullable 
@Value.Auxiliary
@Value.Redacted
public @interface MongoId {}
```

Otherwise this sea of annotations appears multiple times within my code. It works for Jackson because of `@JacksonAnnotationsInside`.

Is there some equivalent way to have an annotation that contains other annotations for Immutables?

**Comment by elucash** [comment]
We use some meta annotation functionality for styles and some other type level stuff and even separate [annotation-injection](https://github.com/immutables/immutables/blob/master/annotate/src/org/immutables/annotate/InjectAnnotation.java) mechanism (this could allow inserting many class/runtime annotation, including `@JsonProperty("[[]]")`). But for the particular use case of yours, where we need to activate compile-time annotations like `@Value.Auxiliary` and `@Value.Redacted` (and `@Nullable`) separate pathway need to be implemented.

**Comment by msmerc** [comment]
Many thanks for your reply!

It seems the modern way is annotations everywhere. If I can roll my own annotations that eloquently contain the annotations from a smattering of 3rd party libraries it makes the code much more readable - would be a helpful feature to add!

**Comment by elucash** [comment]
Yeah, it's like we've tried to minimize annotation usage (with styles & meta-style annotations) but there's still plenty of them and we've contributed some too 🤷‍♂️. There a some conceptual problems associated too. First, that I already mentioned, we can  implant third party annotations into the generated code, but not group our own feature annotations (styles are quite different, meta styles and package / parent-package styles seems to work ok). The second problem is that some of the annotations are not applicable to annotations, so we need to use string code snippets or class-literal to describe specifically `@Target`ed annotation to be applied. There are also implementation problems which are solvable, of course, but could be daunting to code.
